core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -164,7 +164,15 @@ private[spark] class SparkSubmit extends Logging {
             }
         }
       } else {
-        runMain(childArgs, childClasspath, sparkConf, childMainClass, args.verbose)
+        if (sparkConf.getOption("spark.kubernetes.kerberos.proxyUser").isDefined) {
+          // scalastyle:off println
+          printStream.println("Running as proxy user in k8s cluster mode...")
+          // scalastyle:on println
+          SparkHadoopUtil.get.runAsSparkUser(
+            () => runMain(childArgs, childClasspath, sparkConf, childMainClass, args.verbose))
```

Comment:
So if I have followed the code flow correctly now when using Kerberos auth the value of `SPARK_USER` provided to the driver is the user we're launching the job as from a Spark/HDFS perspective?